src/pages/home/report/ReportActionsView.tsx

```diff
@@ -49,34 +52,63 @@ type ReportActionsViewProps = ReportActionsViewOnyxProps & {
 
     /** The report actions are loading newer data */
     isLoadingNewerReportActions?: boolean;
+
+    /** Whether the report is ready for comment linking */
+    isReadyForCommentLinking?: boolean;
 };
 
+const DIFF_BETWEEN_SCREEN_HEIGHT_AND_LIST = 120;
+const SPACER = 16;
+
+let listOldID = Math.round(Math.random() * 100);
+
 function ReportActionsView({
     report,
     session,
     parentReportAction,
-    reportActions = [],
+    reportActions: allReportActions = [],
     isLoadingInitialReportActions = false,
     isLoadingOlderReportActions = false,
     isLoadingNewerReportActions = false,
+    isReadyForCommentLinking = false,
 }: ReportActionsViewProps) {
     useCopySelectionHelper();
     const reactionListRef = useContext(ReactionListContext);
+    const route = useRoute<RouteProp<CentralPaneNavigatorParamList, typeof SCREENS.REPORT>>();
+    const reportActionID = route?.params?.reportActionID;
     const didLayout = useRef(false);
     const didSubscribeToReportTypingEvents = useRef(false);
-    const isFirstRender = useRef(true);
-    const hasCachedActions = useInitialValue(() => reportActions.length > 0);
-    const mostRecentIOUReportActionID = useMemo(() => ReportActionsUtils.getMostRecentIOURequestActionID(reportActions), [reportActions]);
+
+    // triggerListID is used when navigating to a chat with messages loaded from LHN. Typically, these include thread actions, task actions, etc. Since these messages aren't the latest,we don't maintain their position and instead trigger a recalculation of their positioning in the list.
+    // we don't set currentReportActionID on initial render as linkedID as it should trigger visibleReportActions after linked message was positioned
+    const [currentReportActionID, setCurrentReportActionID] = useState('');
+    const isFirstLinkedActionRender = useRef(true);
+
     const network = useNetwork();
-    const {isSmallScreenWidth} = useWindowDimensions();
+    const {isSmallScreenWidth, windowHeight} = useWindowDimensions();
+    const contentListHeight = useRef(0);
+    const isFocused = useIsFocused();
     const prevNetworkRef = useRef(network);
     const prevAuthTokenType = usePrevious(session?.authTokenType);
-
+    const [isNavigatingToLinkedMessage, setNavigatingToLinkedMessage] = useState(!!reportActionID);
     const prevIsSmallScreenWidthRef = useRef(isSmallScreenWidth);
-
-    const isFocused = useIsFocused();
     const reportID = report.reportID;
-    const hasNewestReportAction = reportActions[0]?.isNewestReportAction;
+    const isLoading = (!!reportActionID && isLoadingInitialReportActions) || !isReadyForCommentLinking;
```

Comment:
@perunt Coming from [this issue](https://github.com/Expensify/App/issues/40937), we're considering a [proposal](https://github.com/Expensify/App/issues/40937#issuecomment-2106180673) that would change the `isLoading` property to:

```
const isLoading = !!reportActionID && isLoadingInitialReportActions && !isReadyForCommentLinking;
```

This is in order to stop setting the `reportActions` array to empty here

https://github.com/Expensify/App/blob/2a6d63d7f17b1e4fba94f26103f13c6afe294b91/src/pages/home/report/ReportActionsView.tsx#L165-L167

for cases where we already have the whole report and actions available in Onyx.

My problem is that I am a bit unsure what `isLoadingInitialReportActions` part is actually doing here. Are you able to explain why it's needed and/or what case it handles? 

My guess is that it shouldn't be true if we've already opened the report and are just returning to it, i.e.

1. Open some report (non-comment linked, so `reportActionID` evaluates as falsey)
2. Navigate away and then return to that report via a comment link, so `reportActionID` will be true
3. `isLoadingInitialReportActions` should not be true as we already have all the `reportActions` from (1), so `isLoading` will be false (assuming `isReadyForCommentLinking` is true).

Reply:
@perunt bump, what do you think about the above proposal?

Reply:
Looks okay to me, but we need to be sure that we don't skip the first frame of rendering, as the whole positioning would be glitching. So, for testing this proposal, please use the test steps from this PR as well